### DISCUSSION TITLE
Sorting child features

### DISCUSSION THREAD

**Discussion by BoswachterMarc** [opengisch/QField#4839] [Ideas]
When adding child features in a one to many relation, there is no possibility to sort those features on the child or many side/form. By default, they are sorted from old to new (ascending) while sorting them the other way from new to old (descending) is often more desirable.
In QGis it is possible to change the sort order, but that order is not preserved in QField.
Could this be a improvement?
![Screenshot_20231210_215856](https://github.com/opengisch/QField/assets/57868975/995c5bd2-d1e0-4245-89c5-6219719ccc35)

**Comment by nirvn** [comment]
@BoswachterMarc , that would be a good enhancement to have indeed. On the technical side of things, it's quite doable; let us know if you are aware of someone able to sponsor this.

**Reply by BoswachterMarc** [reply]
I'm afraid that I don't know a sponsor and that a personal, limited contribution will not be enough...

**Reply by nirvn** [reply]
@BoswachterMarc , you've seeded the idea, that's a good start :)

**Reply by ZsanettMed** [reply]
Hello @BoswachterMarc, I have shared your idea on our revitalized [QField Idea Platform](https://ideas.qfield.org/app-feature-requests/p/sorting-child-features). This way, you and other users can upvote it and help bring it to life. :) Thank you for your contribution!

**Comment by m-kuhn** [comment]
QField supports [ordered relation editor](https://github.com/opengisch/qgis-ordered-relation-editor). This is optimized for manual sorting of child items and what you are looking for sounds like automatic sorting but it could still be worth a try.

**Reply by BoswachterMarc** [reply]
Thank you @m-kuhn. I have been looking at it, but after re-ordering, when you do other operations in the QGis project, the order is not saved. In other words, the automatic sorting on the QField side would be a nice enhancement ;-)

[Screencast from 21-12-23 12:51:34.webm](https://github.com/opengisch/QField/assets/57868975/e75cf8bb-aa54-4b34-b516-0b4990c7f7db)

There is also a issue with the plug-in; I will give feedback to the developer of that plug-in.
